[Failing-test] Kubelet panic in NodeDeclareFeatures code

### Which jobs are failing?

alpha jobs

### Which tests are failing?

Entire job.


### Since when has it been failing?

Test grid for some reason doesn't go very far back.

### Testgrid link

https://testgrid.k8s.io/sig-node-containerd#ci-node-e2e-containerd-alpha-features

### Reason for failure (if possible)

```
I0105 07:36:00.527576    2667 kubelet.go:2611] "SyncLoop DELETE" source="api" pods=["mirror-pod-6526/static-pod-909c75fb-fb8e-4443-b1d1-79f09ce6219a-tmp-node-e2e-9ee50a67-ubuntu-gke-2404-1-34-amd64-v20251218a"]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d6e0b9]

goroutine 234 [running]:
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).HandlePodUpdates(0xc000359008, {0xc00112cfa8, 0x1, 0x0?})
	k8s.io/kubernetes/pkg/kubelet/kubelet.go:2830 +0x459
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoopIteration(0xc000359008, {0x295a2d0, 0x3b54280}, 0xc0000f5ce0, {0x2963408, 0xc000359008}, 0xc00043ee70, 0xc00043ef50, 0xc00033ff10)
	k8s.io/kubernetes/pkg/kubelet/kubelet.go:2613 +0xa22
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncLoop(0xc000359008, {0x295a2d0, 0x3b54280}, 0xc0000f5ce0, {0x2963408, 0xc000359008})
	k8s.io/kubernetes/pkg/kubelet/kubelet.go:2542 +0x34c
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).Run(0xc000359008, 0xc0000f5ce0)
	k8s.io/kubernetes/pkg/kubelet/kubelet.go:1896 +0x8db
created by k8s.io/kubernetes/cmd/kubelet/app.startKubelet in goroutine 1
	k8s.io/kubernetes/cmd/kubelet/app/server.go:1264 +0xa9
```

### Anything else we need to know?

https://github.com/kubernetes/kubernetes/blob/b2ac9e206fdd912f35f2ab5b3c5b5243303ba14b/pkg/kubelet/kubelet.go#L2830

Looks like oldPod may not always be set so this triggers the panic.

### Relevant SIG(s)

/sig node

--- a/pkg/kubelet/kubelet.go
+++ b/pkg/kubelet/kubelet.go
@@ -2826,7 +2826,7 @@ func (kl *Kubelet) HandlePodUpdates(pods []*v1.Pod) {
 			}
 		}
 
-		if utilfeature.DefaultFeatureGate.Enabled(features.NodeDeclaredFeatures) {
+		if utilfeature.DefaultFeatureGate.Enabled(features.NodeDeclaredFeatures) && oldPod != nil {
 			oldPodInfo := &ndf.PodInfo{Spec: &oldPod.Spec, Status: &oldPod.Status}
 			newPodInfo := &ndf.PodInfo{Spec: &pod.Spec, Status: &pod.Status}
 			reqs, err := kl.nodeDeclaredFeaturesFramework.InferForPodUpdate(oldPodInfo, newPodInfo, kl.version)

--- a/pkg/kubelet/kubelet_test.go
+++ b/pkg/kubelet/kubelet_test.go
@@ -4760,6 +4760,8 @@ func TestSyncPodNodeDeclaredFeaturesUpdate(t *testing.T) {
 		name               string
 		featureGateEnabled bool
 		nodeFeatures       []string
+		oldPod             *v1.Pod
+		newPod             *v1.Pod
 		registeredFeatures []ndf.Feature
 		expectEvent        bool
 		expectedEventMsg   string
@@ -4769,6 +4771,8 @@ func TestSyncPodNodeDeclaredFeaturesUpdate(t *testing.T) {
 			name:               "Feature gate disabled",
 			featureGateEnabled: false,
 			componentVersion:   "1.35.0",
+			oldPod:             oldPod,
+			newPod:             newPod,
 			registeredFeatures: []ndf.Feature{},
 			expectEvent:        false,
 		},
@@ -4776,6 +4780,8 @@ func TestSyncPodNodeDeclaredFeaturesUpdate(t *testing.T) {
 			name:               "Feature gate enabled, no requirements from update",
 			featureGateEnabled: true,
 			componentVersion:   "1.35.0",
+			oldPod:             oldPod,
+			newPod:             newPod,
 			nodeFeatures:       []string{"FeatureA"},
 			registeredFeatures: []ndf.Feature{createMockFeature(t, "FeatureA", false, "")},
 			expectEvent:        false,
@@ -4784,6 +4790,8 @@ func TestSyncPodNodeDeclaredFeaturesUpdate(t *testing.T) {
 			name:               "Feature gate enabled, requirements met",
 			featureGateEnabled: true,
 			componentVersion:   "1.35.0",
+			oldPod:             oldPod,
+			newPod:             newPod,
 			nodeFeatures:       []string{"FeatureA"},
 			registeredFeatures: []ndf.Feature{createMockFeature(t, "FeatureA", true, "")},
 			expectEvent:        false,
@@ -4792,6 +4800,8 @@ func TestSyncPodNodeDeclaredFeaturesUpdate(t *testing.T) {
 			name:               "Feature gate enabled, requirements not met",
 			featureGateEnabled: true,
 			componentVersion:   "1.35.0",
+			oldPod:             oldPod,
+			newPod:             newPod,
 			nodeFeatures:       []string{"FeatureB"},
 			registeredFeatures: []ndf.Feature{createMockFeature(t, "FeatureA", true, "")},
 			expectEvent:        true,
@@ -4801,8 +4811,20 @@ func TestSyncPodNodeDeclaredFeaturesUpdate(t *testing.T) {
 			name:               "feature generally available - not declared",
 			featureGateEnabled: true,
 			componentVersion:   "1.35.0",
+			oldPod:             oldPod,
+			newPod:             newPod,
 			nodeFeatures:       []string{""},
 			registeredFeatures: []ndf.Feature{createMockFeature(t, "FeatureA", true, "1.34.0")},
+			expectEvent:        false,
+		},
+		{
+			name:               "Feature gate enabled, old pod state missing, node declared feature check skipped",
+			featureGateEnabled: true,
+			componentVersion:   "1.35.0",
+			oldPod:             nil,
+			newPod:             newPod,
+			nodeFeatures:       []string{"FeatureB"},
+			registeredFeatures: []ndf.Feature{createMockFeature(t, "FeatureA", true, "")},
 			expectEvent:        false,
 		},
 	}
@@ -4826,10 +4848,12 @@ func TestSyncPodNodeDeclaredFeaturesUpdate(t *testing.T) {
 			kubelet.nodeDeclaredFeaturesSet = ndf.NewFeatureSet(kubelet.nodeDeclaredFeatures...)
 			kubelet.version = utilversion.MustParse("1.35.0")
 
-			kubelet.podManager.SetPods([]*v1.Pod{oldPod})
-			kubelet.statusManager.SetPodStatus(klog.TODO(), newPod, v1.PodStatus{Phase: v1.PodRunning})
-			kubelet.HandlePodUpdates([]*v1.Pod{newPod})
+			if tc.oldPod != nil {
+				kubelet.podManager.SetPods([]*v1.Pod{tc.oldPod})
+			}
 
+			kubelet.statusManager.SetPodStatus(klog.TODO(), tc.newPod, v1.PodStatus{Phase: v1.PodRunning})
+			kubelet.HandlePodUpdates([]*v1.Pod{tc.newPod})
 			if tc.expectEvent {
 				select {
 				case event := <-recorder.Events:

